core/lib/instrument.c

```diff
@@ -1462,7 +1465,7 @@ instrument_thread_exit(dcontext_t *dcontext)
 #    endif
 
 #    ifdef DEBUG
-    /* PR 470957: avoid racy crashes by not freeing in release build */
+    /* i#271: avoid racy crashes by not freeing in release build. */
```

Comment:
Your commit message suggests that these changes are done for release builds but here we have an ifdef DEBUG.

https://github.com/DynamoRIO/dynamorio/pull/4350#issue-443017097

Reply:
Actually the commit message lists 3 other different places where the freeing is moved to release build, and has a separate list item for this one: "Moves client_data_t, client_todo_list_t, and thread-private client_flush_req_t to be PROTECTED to make it thread-local and thus not need freeing (the free is avoided for i#271).".  I will make it clearer in the message that this stays DEBUG-only.